frame/nft-fractionalisation/src/lib.rs

```diff
@@ -142,122 +166,97 @@ pub mod pallet {
 		#[pallet::weight(10_000 + T::DbWeight::get().writes(2).ref_time())]
 		/// Pallet's account must be funded before lock is possible!
 		/// 5EYCAe5gjC5dxKPbV2GPQUetETjFNSYZsSwSurVTTXidSLbh
-		pub fn lock_nft_create_asset(
+		pub fn fractionalise(
 			origin: OriginFor<T>,
-			collection_id: T::CollectionId,
-			item_id: T::ItemId,
+			nft_collection_id: T::NftCollectionId,
+			nft_id: T::NftId,
 			asset_id: AssetIdOf<T>,
 			beneficiary: T::AccountId,
-			min_balance: AssetBalanceOf<T>,
-			amount: AssetBalanceOf<T>,
+			fractions: AssetBalanceOf<T>,
 		) -> DispatchResult {
-			let _who = ensure_signed(origin)?;
-			let admin_account_id = Self::pallet_account_id();
-
-			Self::do_lock_nft(collection_id, item_id)?;
-			Self::do_create_asset(asset_id, admin_account_id, min_balance)?;
-			Self::do_mint_asset(asset_id, &beneficiary, amount)?;
-
-			// Mutate this storage item to retain information about the amount minted.
-			<AssetsMinted<T>>::try_mutate(
-				asset_id,
-				|assets_minted| -> Result<(), DispatchError> {
-					match assets_minted.is_some() {
-						true =>
-							*assets_minted = Some(assets_minted.unwrap().saturating_add(amount)),
-						false => *assets_minted = Some(amount),
-					}
-
-					Ok(())
-				},
-			)?;
-
-			// Mutate this storage item to retain information about the asset created.
-			<AssetToNft<T>>::try_mutate(asset_id, |nft_id| -> Result<(), DispatchError> {
-				*nft_id = Some((collection_id, item_id));
+			let who = ensure_signed(origin)?;
+			let nft_owner =
+				T::Nfts::owner(&nft_collection_id, &nft_id).ok_or(Error::<T>::NoPermission)?;
+			ensure!(nft_owner == who, Error::<T>::NoPermission);
```

Comment:
Do we actually need this check? If the caller of the extrinsic is not the NFT owner, the `do_lock_nft` function would throw an error too

Reply:
It won't throw an error, the `do_lock_nft` method simply applies the lock without doing any checks